lib/makara/context.rb

```diff
@@ -1,83 +1,144 @@
 require 'digest/md5'
 
-# Keeps track of the current and previous context (hexdigests)
-# If a new context is needed it can be generated via Makara::Context.generate
-
+# Keeps track of the current stickiness state for different Makara configurations
 module Makara
```

Comment:
Good point. Maybe having a new class, `Makara::CookieParser` (or `Makara::CookieSerializer`, or `Makara::ContextCookie`... or just `Makara::Cookie`, I think that'd work), that knows how to build the hash from the cookie string and how to serialize the hash), and the whole thing would be:
- The middleware knows the cookie name, gets the contents from the request, uses that new class to build the hash, and passes that to `init`.
- The `commit` method is moved to the middleware, that gets what should be committed from `Context`, uses that new class to build the cookie string, and then modifies the headers.

I'll try that to see how it looks!

Reply:
Hmm... although, thinking about it again, if the new class is called `Makara::Cookie`, it would imply it is who knows about the cookie name and it would be the one initialising the context and _committing_ the new cookie. If we want to keep that in the middleware, perhaps `Makara::ContextParser` or `Makara::ContextSerializer` or something like that would be better, for parsing the cookie contents, and for serialising them.